pkg/collector/corechecks/snmp/devicecheck/devicecheck.go

```diff
@@ -0,0 +1,160 @@
+package devicecheck
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/checkconfig"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/fetch"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/metadata"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/report"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/session"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/valuestore"
+)
+
+const (
+	snmpLoaderTag    = "loader:core"
+	serviceCheckName = "snmp.can_check"
+)
+
+// DeviceCheck hold info necessary to collect info for a single device
+type DeviceCheck struct {
+	config  *checkconfig.CheckConfig
+	sender  *report.MetricSender
+	session session.Session
+}
+
+// NewDeviceCheck returns a new DeviceCheck
+func NewDeviceCheck(config *checkconfig.CheckConfig, ipAddress string) (*DeviceCheck, error) {
```

Comment:
Wonder if this should only accept the `CheckConfig`, and it should be up to the caller to create the configuration properly. Imagine having new props that change from one config to the next (for example, MAC address, subnet), it would require changing the signature.

Reply:
Currently, the IP has a specific meaning compared to (for example, MAC address, subnet). 
The IP is the only thing change changes when we need to call `NewDeviceCheck` for agent corecheck autodiscovery.

For now, I don't see fields other than IP that need to be changed.

It's possible to extract the logic outside, but I think it's not necessarily better.

Ideally, I wanted to not use `checkconfig.CheckConfig` and have a separate `DeviceConfig` structure that only holds configs for a single device (e.g. without network configs). But that would require additional changes and might lead to some duplicate with existing `checkconfig.CheckConfig`.

Reply:
I think `DeviceConfig` should still need networks info for NDM

Reply:
> I think DeviceConfig should still need networks info for NDM

Yes, agree the subnet name. But I think other configs like `workers`, `discovery_workers`, allowed failures, etc, are not needed.